src/xdmf_output.c

```diff
@@ -64,7 +64,7 @@ static PetscErrorCode WriteFieldData(DM dm, Vec global_vec, PetscViewer viewer)
     PetscCall(PetscFree(comp));
   }
 
-  if (use_natural) PetscCall(VecDestroy(&data_vec));  // FIXME: does this leave an extra ref in non-natural case?
+  PetscCall(VecDestroy(&data_vec));  // FIXME: does this leave an extra ref in non-natural case?
```

Comment:
If I understand this question correctly, the "extra ref" you're referring to is to `global_vec` in the case of a refined mesh. The lifetime of `global_vec` exceeds this function, so it should still have a live reference anyway. Does that sound right to you?